### PR TITLE
FIX : sequential pipeline image order

### PR DESCRIPTION
**Pull request by kartik-579**
# Description

In the previous release for sequential pipeline, cd artifact images were not showing in order. Added sorting of ci artifacts on the basis of their creation time

### CODE REVIEW

File: pkg/pipeline/PipelineBuilder.go [hunk 1]
```diff
@@ -1671,14 +1687,15 @@ func (impl PipelineBuilderImpl) BuildArtifactsForCdStage(pipelineId int, stageTy
...
-					ScanEnabled:     wfr.CdWorkflow.CiArtifact.ScanEnabled,
+					Id:                     wfr.CdWorkflow.CiArtifact.Id,
+					Image:                  wfr.CdWorkflow.CiArtifact.Image,
+					ImageDigest:            wfr.CdWorkflow.CiArtifact.ImageDigest,
+					MaterialInfo:           mInfo,
+					RunningOnParent:        runningOnParent,
+					Latest:                 latest,
+					Scanned:                wfr.CdWorkflow.CiArtifact.Scanned,
+					ScanEnabled:            wfr.CdWorkflow.CiArtifact.ScanEnabled,
+					CiArtifactCreationTime: wfr.CdWorkflow.CiArtifact.CreatedOn,
```
Comment (pghildiyal) [inline pkg/pipeline/PipelineBuilder.go:1698]: we dont need it in response

File: pkg/pipeline/PipelineBuilder.go [hunk 2]
```diff
@@ -1714,12 +1731,13 @@ func (impl PipelineBuilderImpl) BuildArtifactsForCIParent(cdPipelineId int, ciAr
...
-				MaterialInfo: mInfo,
-				ScanEnabled:  artifact.ScanEnabled,
-				Scanned:      artifact.Scanned,
+				Id:                     artifact.Id,
+				Image:                  artifact.Image,
+				ImageDigest:            artifact.ImageDigest,
+				MaterialInfo:           mInfo,
+				ScanEnabled:            artifact.ScanEnabled,
+				Scanned:                artifact.Scanned,
+				CiArtifactCreationTime: artifact.CreatedOn,
```
Comment (pghildiyal) [inline pkg/pipeline/PipelineBuilder.go:1740]: we dont need it in response

File: pkg/pipeline/PipelineBuilder.go [hunk 3]
```diff
@@ -1626,6 +1636,12 @@ func (impl PipelineBuilderImpl) GetArtifactsForCdStage(cdPipelineId int, parentI
 		impl.logger.Errorw("error in getting artifacts for cd", "err", err, "parentStage", parentType, "stage", stage)
 		return ciArtifactsResponse, err
 	}
+	//sorting ci artifacts on the basis of creation time
+	if ciArtifacts != nil && len(ciArtifacts) > 0 {
```
Comment (pghildiyal) [inline pkg/pipeline/PipelineBuilder.go:1640]: len check not needed

File: internal/sql/repository/CiArtifactRepository.go [hunk 4]
```diff
@@ -115,7 +115,7 @@ func (impl CiArtifactRepositoryImpl) GetArtifactsByCDPipeline(cdPipelineId int)
 	var artifactsAB []CiArtifact
 
 	queryFetchArtifacts := ""
-	queryFetchArtifacts = "SELECT cia.id, cia.data_source, cia.image, cia.image_digest FROM ci_artifact cia" +
+	queryFetchArtifacts = "SELECT cia.id, cia.data_source, cia.image, cia.image_digest, cia.created_on FROM ci_artifact cia" +
```
Comment (pghildiyal) [inline internal/sql/repository/CiArtifactRepository.go:118]: remove created_on, we will sort by id

File: pkg/pipeline/PipelineBuilder.go [hunk 5]
```diff
@@ -1626,6 +1636,12 @@ func (impl PipelineBuilderImpl) GetArtifactsForCdStage(cdPipelineId int, parentI
...
+		sort.SliceStable(ciArtifacts, func(i, j int) bool {
+			return ciArtifacts[i].CiArtifactCreationTime.After(ciArtifacts[j].CiArtifactCreationTime)
```
Comment (pghildiyal) [inline pkg/pipeline/PipelineBuilder.go:1642]: sort by id desc

File: internal/sql/repository/CiArtifactRepository.go [hunk 6]
```diff
@@ -189,7 +189,7 @@ func (impl CiArtifactRepositoryImpl) GetArtifactsByCDPipeline(cdPipelineId int)
 			}
 		}
 		artifactsAll = append(artifactsAll, a)
-		if len(artifactsAll) >= 10 {
+		if len(artifactsAll) >= 30 {
```
Comment (pghildiyal) [inline internal/sql/repository/CiArtifactRepository.go:192]: should we take this number from outside

Comment (kartik-579) [reply internal/sql/repository/CiArtifactRepository.go:192]: in request parameter? we can, but that will add work on frontend side. Maybe we can update this in future release

File: pkg/pipeline/PipelineBuilder.go [hunk 7]
```diff
@@ -1648,7 +1664,7 @@ func (impl PipelineBuilderImpl) BuildArtifactsForParentStage(cdPipelineId int, p
 
 func (impl PipelineBuilderImpl) BuildArtifactsForCdStage(pipelineId int, stageType bean2.WorkflowType, ciArtifacts []bean.CiArtifactBean, artifactMap map[int]int, parent bool) ([]bean.CiArtifactBean, map[int]int, error) {
```
Comment (pghildiyal) [inline pkg/pipeline/PipelineBuilder.go:1665]: pass 30 for the calling function, both to this and also to BuildArtifactsForCIParent

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**